features/202-session-management/config.md

```diff
@@ -0,0 +1,51 @@
+# Session Configuration
+
+## Sample configuration
+
+In user configuration:
+```yaml
+clients:
+    - name: Web App
+      disabled: false
+      api_key: XXX
+      session_transport: cookie
+    - name: iOS App
+      disabled: false
+      api_key: YYY
+      session_transport: header
+    - name: Android App
+      disabled: false
+      api_key: ZZZ
+      session_transport: header
+session:
+    access_token_lifetime: 1800
+    refresh_token_disabled: false
+    refresh_token_lifetime: 86400
+    session_idle_timeout_enabled: true
+    session_idle_timeout: 300
+```
+
+## User Configuration
+
+The value of key `clients` is a list of client configuration:
+- `name`: Name of client. Show in UI (e.g. portal)
```

Comment:
I don't think the name of client shall be determined by API key and show in UI portal.

We should allow developers to use the same API key in different clients, and each client can set the name of current session.

The name in UI should be determined by client side because the use case of a "named session" is for end-user to identify where the session come from. The name would probably be something like "Ben Cheng's iPhone", or "Ben Cheng's 2nd Blackberry at work" instead of a general name for each token.

Reply:
So `name` is a developer-facing name. The display name of a session should be derived from the new session properties. For example, for our iOS SDK, we should get the model of the device and store it in the session properties. We need to come up with a list of properties we can possibly get from different platform and specify them.

Reply:
As mentioned in design doc, 'Client' means client application, e.g. web app and mobile app. The name would be just an internal name shown in **developer** portal.

If developer want to name a session, they could set it as custom attributes of session.

Reply:
> example, for our iOS SDK, we should get the model of the device and store it in the session properties.
This would be the user agent of user.

Reply:
> This would be the user agent of user.

What value do we set it to? Can we specify this?

Reply:
I don't know what is the usage of having a Client name in Developer Portal to name a session, why it exists at all?

It would make more senses if we rename `clients` to `api_keys`, aka, something like this:

```
api_keys:
    - name: Default API Key
       disabled: false
       key: XXX
    - name: Employee App API Key
       disabled: false
       key: XXX
```

(However, I feel like it is out of scope and very confusing, we're talking about sessions management here right? Why bother we support multiple API keys suddenly?)

For Client's session name, we need to support session name to display in Developer Portal and when a user needs to see it in their applications (say they have a page to list all sessions), both names need to be identical. It shouldn't be a custom attributes. That's at least useful for things like doing customer technical support, or regular security audit.

Regarding @louischan-oursky 's suggestion, that make sense, but that should just be a feature in Client SDK to show the "default session name", each session name still need to be changable by clients side.

Reply:
Let's clarify the scope of 'session management'. My understanding is that it includes how session works and how to manipulate sessions.

I don't see much app supporting naming individual sessions. Most UI display attributes derived from user agent & IP, such as device type, app version, IP address, country, etc. Do you have a use case requiring native support of session names?

Reply:
I have used multiple apps support naming individual sessions. I can try to list them out with research but maybe you can do the research instead? (E.g. iClouds allow naming of each devices, which effectively is used as session name)

Reply:
Naming device is indeed a more common feature. However session are much more transient than device, so I don't see usefulness in naming sessions.

I'd expect device name to be more persistent than session name. For example, Device name should remains the same when I re-install the app. It may require using native API from mobile devices, and tracking across installation would be likely against OS policy.

Developer can implement session name if required using custom attributes. Do you have concern on using custom attributes for session name?

Reply:
As explained above, because we will need to display the session name in Developer Portal, I don't think it is suitable to use custom attributes for that. It is common enough to be built in.

I haven't seen anyone implemented a "persistent device name" in session management feature actually. E.g. no matter it is Google or iCloud, I can freely rename my device name whatever, and I will see the device name in my session lists. In fact I can have multiple sessions with one devices (say an Android devices with multiple apps that connected with the same services) and have different names.

I don't think we have the concept of devices here yet? and I assume practically we don't really track "Device name", but "session name" only.

Device tracking is usually more related with things like ad-tracking etc, which have privacy concern and not really related with this scope. The device name mentioned above is just to make a sensible default for session name, not for persistent tracking purpose.

Reply:
Personally I do not found session name a feature common enough - I have not yet uses app that allows assigning name to individual sessions. For developer portal, I expect it to show a full list of custom attributes for each session, so putting session name there if needed should be usable.

For iCloud, it uses device name, which is persistent across session, and in iOS this is available as `UIDevice.name`. However, there seems no consistent way to get device name on Android, and impossible on web platform. Since we cannot provide a sensible default for session name, I don't think we should support this natively.

Reply:
Despite your personal preference, this feature is common enough. However I think this should be discussed in another part of the spec, will create another comment there. Thanks.